platform/android/java/lib/src/org/godotengine/godot/input/GodotInputHandler.java

```diff
@@ -484,8 +484,17 @@ static boolean handleMouseEvent(final MotionEvent event) {
 		final float tiltX = (float)-Math.sin(orientation) * tiltMult;
 		final float tiltY = (float)Math.cos(orientation) * tiltMult;
 
-		final float verticalFactor = event.getAxisValue(MotionEvent.AXIS_VSCROLL);
-		final float horizontalFactor = event.getAxisValue(MotionEvent.AXIS_HSCROLL);
+		float verticalFactor = 0;
+		float horizontalFactor = 0;
+
+		if (event.isFromSource(InputDevice.SOURCE_ROTARY_ENCODER)) {
+			// If event came from RotaryEncoder (Bezel or Crown rotate event on Wear OS smart watches),
+			// convert it to vertical mouse wheel event.
+			verticalFactor = event.getAxisValue(MotionEvent.AXIS_SCROLL);
```

Comment:
The documentation you linked to has additional logic that is not replicated here; is there a reason to leave that logic out?

You should also provide the ability for users to select whether this should be converted to a vertical event or to a horizontal event using project settings.

Reply:
Thank you for review. Yes you are right I need to make negation for axis value. All other logic in [this documentation](https://developer.android.com/training/wearables/user-input/rotary-input#java) connected with getting some settings from configuration, like getting scroll velocity and applying RotaryInput to UI control.

> You should also provide the ability for users to select whether this should be converted to a vertical event or to a horizontal event using project settings.

I have concerns about ability to change direction of this input from vertical to horizontal through project settings. As I understand we should convert some specific inputs to another more general inputs. So here I conver RotaryInput to mouse wheel input(most of miсe have only one wheel for vertical scrolling). If I added such settings It would be very disappointed, because in the same game running on PC mouse wheel would cause vertical scrolling but on smart watches crown wheel or bezel wheel would cause horizontal scrolling. Anyway In the same game such event can cause as vertical as horizontal scrolling in different scenes. So how to handle this events and which axis should be chaned by this event should be defined in specific places(inside controls or scene or other object scripts) in my opinion.
So if I have not convinced you, let me know and I will add this setting.

Reply:
For a cross-platform game/app, I'd agree that consistency is key and for that reason the default value for the project setting should be set to vertical to simulate similar interaction on other platforms. 

However a developer can also decide to target wear os devices only and as such should not be restricted by other platforms conventions. For that reason, we should provide the option to override how the rotary input is interpreted.

Reply:
Added option to change RotaryInpur axis through project settings.